### PR TITLE
Update documentation to reflect supported versions

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**
This PR relates to the effort to document supported language versions.
Please refer to this [document](https://docs.google.com/spreadsheets/d/1-ZeS-sg22A4QJu9xWzi_XfVYvmEHgHwV-4Cxu9jlkqM/edit#gid=0) for further information.

### CODE REVIEW

File: sonar-csharp-plugin/src/main/resources/documentation.md [hunk 1]
```diff
@@ -7,6 +7,13 @@ key: csharp
 <!-- update_center:csharp -->
 <!-- /static -->
 
+## Supported Versions
```
Comment (pavel-mikula-sonarsource) [inline sonar-csharp-plugin/src/main/resources/documentation.md:10]: As a concept, isn't this a pointless change given the fact that documentation lives only in Documentation team CMS and there's no more synchronization in place?

On the long run, these files should get dropped from all analyzers.

Comment (nils-werner-sonarsource) [reply sonar-csharp-plugin/src/main/resources/documentation.md:10]: Pavel, that's a good point, which we also discussed in the small circle with the PM and documentation team. We came to the conclusion that the embedded documentation is a good source of truth not only for the supported version but also for other content that can be extracted there by the documentation team. By being part of the Git history, it is also easy to keep track of how the language version support has evolved.

Comment (pavel-mikula-sonarsource) [reply sonar-csharp-plugin/src/main/resources/documentation.md:10]: The point is that these files will get dropped sooner or later in any case. So there will be no history.

Comment (nils-werner-sonarsource) [reply sonar-csharp-plugin/src/main/resources/documentation.md:10]: Do you have any source for the statement? The documentation team is about to discover new ways to implement automation again.

Comment (pavel-mikula-sonarsource) [reply sonar-csharp-plugin/src/main/resources/documentation.md:10]: Just the combination of facts that this is the old way of doing this and the ultimate source of documentation truth lives in the new documentation system maintained by the docs team.
So trying to automate this again sounds like a step backwards that doesn't make much sense now.

**Comment by pavel-mikula-sonarsource** [comment]
What is your plan with this PR? It's hanging in our Kanban / "In progress" column.

The document you've linked above doesn't explain much. 
It's not clear to me what "supported" means. 

For C#:
* We analyze C# 11 code (latest release) 
* Our existing rules have C# 11 specific logic, so old rules are C# 11 aware to avoid most of FPs/FNs there.
* There are no new rules specific to C# 11 features (same for C# 10, 9, 8 I think, maybe except very few exceptions).
* Once C# 12 is released (expected Q4 2023), the analysis will just work out of the box also with old releases of the analyzer. We expect new FPs/FNs on C# 12 specific syntaxes.

**Comment by nils-werner-sonarsource** [comment]
> What is your plan with this PR? It's hanging in our Kanban / "In progress" column.
> 
> The document you've linked above doesn't explain much. It's not clear to me what "supported" means.
> 
> For C#:
> 
> * We analyze C# 11 code (latest release)
> * Our existing rules have C# 11 specific logic, so old rules are C# 11 aware to avoid most of FPs/FNs there.
> * There are no new rules specific to C# 11 features (same for C# 10, 9, 8 I think, maybe except very few exceptions).
> * Once C# 12 is released (expected Q4 2023), the analysis will just work out of the box also with old releases of the analyzer. We expect new FPs/FNs on C# 12 specific syntaxes.

Thank you for providing the data. Based on your feedback we reevaluated the approach to collect and track the data.

**Comment by pavel-mikula-sonarsource** [comment]
Thanks!

**Pull request closed without merging**